### PR TITLE
Initial version

### PR DESCRIPTION
**Pull request by NickGrantONS**
First hack at the DDL repo

### CODE REVIEW

**Comment by NickGrantONS** [comment]
["Spike" DDL repo for dev SSDC](https://trello.com/c/OVMa2ZTf/2357-spike-ddl-repo-for-dev-ssdc)

**Review by gavinleeedwards — CHANGES_REQUESTED** [review]
Can we just go back to having a case schema (rather than casev..), seeing as though ssdc stuff is completely new and separate from Census?  Either way, the permissions still refer to the v2 schema.

**Comment by NickGrantONS** [comment]
> Can we just go back to having a case schema (rather than casev..), seeing as though ssdc stuff is completely new and separate from Census? Either way, the permissions still refer to the v2 schema.

I had some problems with a schema called `case` because it's a reserved word in Postgres 11. Can look into it though if it bothers you a lot @gavinleeedwards 

For now, I've just fixed the permissions scripts.

File: Pipfile [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+flake8 = "*"
+pytest = "==5.4.1"
+pytest-cov = "==2.10.1"
+
+
+[packages]
+psycopg2-binary = "*"
+
+[requires]
+python_version = "3.8"
```
Comment (ryangrundy7) [inline Pipfile:17]: You could make this python 3.9 so we're up to the latest

Comment (NickGrantONS) [reply Pipfile]: done

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+sudo: required
+
+services:
+  - docker
+
+language: python
+
+python:
+  - "3.8"
```
Comment (ryangrundy7) [inline .travis.yml:9]: Bump this as well

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,7 @@
+FROM python:3.8-slim
```
Comment (ryangrundy7) [inline Dockerfile:1]: And this

File: README.md [hunk 4]
```diff
@@ -1 +1,34 @@
-# ssdc-rm-ddl
\ No newline at end of file
+# census-rm-ddl[![Build Status](https://travis-ci.com/ONSdigital/census-rm-ddl.svg?branch=master)](https://travis-ci.com/ONSdigital/census-rm-ddl)
```
Comment (davidmort) [inline README.md:1]: refers to census here

File: README.md [hunk 5]
```diff
@@ -1 +1,34 @@
...
+Contains ground zero scripts to create all the RM database tables, as well as a script to update the DDL from the master branch of the RM components
+
+To run:
+`build_groundzero_ddl.sh`
+
+It's very important that the DDL script to build ground zero is run, instead of manually editing the files in the `groundzero_ddl` directory, so that the database is always perfectly in sync with the code on the master branch.
+
+If you have made changes to Action Scheduler, Case Processor, Exception Manager or the UAC-QID Service and you need to re-generate the ground zero DDL, you can specify specific branches with the following options:
+- `CASE_PROCESSOR_BRANCH`
+- `ACTION_SCHEDULER_BRANCH`
```
Comment (davidmort) [inline README.md:11]: action scheduler doesn't exist anymore so can remove this line

**Review by davidmort — CHANGES_REQUESTED** [review]

File: .travis.yml [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+after_success:
+  - pipenv run codecov
+
+env:
+  global:
+    - PIPENV_IGNORE_VIRTUALENVS=1
+
+branches:
+  only:
+    - master
```
Comment (AdamHawtin) [inline .travis.yml:29]: should be `main` on new repos

File: Pipfile [hunk 7]
```diff
@@ -0,0 +1,17 @@
+[[source]]
+name = "pypi"
+url = "https://pypi.org/simple"
+verify_ssl = true
+
+[dev-packages]
+codecov = "==2.0.22"
+flake8 = "*"
+pytest = "==5.4.1"
+pytest-cov = "==2.10.1"
```
Comment (AdamHawtin) [inline Pipfile:10]: Not sure why these were pinned, these are old versions now.

File: README.md [hunk 8]
```diff
@@ -1 +1,34 @@
...
+
+## Patching a database
+The script `patch_database.py` is used by RM to run database patches from a tagged release of this repository. This script is invoked from our pipelines and will run in a Kubernetes pod to apply any database patches from files in the tagged release version of this repository.
+
+## Manually applied indexes
+[manually_applied_indexes.sql]() Contains indexes we can't or don't want to create in the ground zero scripts. It can be manually run into the database after the main/largest sample load to keep ingest times as fast as possible.
+
+Indexes:
+* postcode_upper_no_space_idx - Index for case/space insensitive postcode search
+
```
Comment (AdamHawtin) [inline README.md:27]: No longer applicable

File: .travis.yml [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+
+python:
+  - "3.9"
+
+install:
+  - pip install pipenv
+  - pipenv install --dev --deploy
+
+script:
+  - PIPENV_PYUP_API_KEY="" pipenv check -i 40291 -i 39253
```
Comment (ryangrundy7) [inline .travis.yml:16]: Don't think we need `PIPENV_PYUP_API_KEY=""` and `-i 40291 -i 39253` now

File: Makefile [hunk 10]
```diff
@@ -0,0 +1,11 @@
+install:
+	pipenv install --dev
+
+check:
+	PIPENV_PYUP_API_KEY="" pipenv check -i 40291
```
Comment (ryangrundy7) [inline Makefile:5]: Don't think we need `PIPENV_PYUP_API_KEY=""` and `-i 40291` now

**Review by ryangrundy7 — COMMENTED** [review]
Just some old pipenv stuff that I think can be removed

File: .travis.yml (hunk 9 above)
Comment (NickGrantONS) [reply .travis.yml:16]: I checked... we do need it

File: Makefile (hunk 10 above)
Comment (NickGrantONS) [reply Makefile:5]: I checked... we do need it

**Review by hbrace — APPROVED** [review]

**Review by liamtoozer — APPROVED** [review]

**Pull request merged**